frontend/src/metabase/query_builder/selectors.js

```diff
@@ -994,3 +994,18 @@ export const getIsEditingInDashboard = state => {
 export const getDashboard = state => {
   return getDashboardById(state, getDashboardId(state));
 };
+
+export const getTemplateTags = createSelector([getCard], card => {
+  return getIn(card, ["dataset_query", "native", "template-tags"]);
+});
```

Comment:
We already have a perfect MLv2 function for this.
`Lib.templateTags(query)` will give you exactly what you need.

In that case, can we not add another selector and use the MLv2 directly in ViewHeader instead?

This is the example for the following SQL:
`select * from PEOPLE where {{state}}` where `state` is a field filter mapped to people.state. It is required with the default value set to `CA`.

```json
{
  "state": {
    "widget-type": "contains",
    "default": [
      "CA"
    ],
    "name": "state",
    "type": "dimension",
    "id": "4200afba-5f26-4d38-bbcd-de6279c7d55e",
    "dimension": [
      "field",
      {
        "base-type": "Text",
        "effective-type": "Text",
        "uuid": "58e74d8d-df71-4485-86f2-c5b616010bb0"
      },
      34
    ],
    "display-name": "State",
    "options": {
      "case-sensitive": false
    },
    "required": true
  }
}
```

Reply:
Actually to use MLv2 is not a bad idea, will do. However, this logic must stay within the `selectors` (or any other than rendering) layer for sure because ideally, our components should be almost purely presentational. Without that, we will never escape this neo-PHP realm where every single thing must be tested with an e2e test or what we call "unit tests".

Reply:
WFM